(F-1207 // B-6.1.5) Error on SMP Only Placing Robot on Docking Station

Can't place a Robot on a Docking Station, works fine in single player.

... 15 more
at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:174)
at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:276)
at cpw.mods.fml.common.asm.transformers.TerminalTransformer.transform(TerminalTransformer.java 14)
at org.objectweb.asm.ClassReader.<init>(Unknown Source)
Caused by: java.lang.NullPointerException
... 13 more
at java.lang.ClassLoader.loadClass(Unknown Source)
at java.lang.ClassLoader.loadClass(Unknown Source)
at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:188)
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.particle.EntityFX
at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685)
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427)
at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547)
at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:335)
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659)
at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165)
at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212)
at net.minecraft.network.play.client.C08PacketPlayerBlockPlacement.func_148833_a(SourceFile:9)
at net.minecraft.network.play.client.C08PacketPlayerBlockPlacement.func_148833_a(SourceFile:60)
at net.minecraft.network.NetHandlerPlayServer.func_147346_a(NetHandlerPlayServer.java:554)
at net.minecraft.server.management.ItemInWorldManager.func_73078_a(ItemInWorldManager.java:376)
at buildcraft.transport.BlockGenericPipe.func_149727_a(BlockGenericPipe.java:798)
at buildcraft.core.ItemRobot.createRobot(ItemRobot.java:43)
java.lang.NoClassDefFoundError: net/minecraft/client/particle/EntityFX

--- a/common/buildcraft/core/robots/EntityRobot.java
+++ b/common/buildcraft/core/robots/EntityRobot.java
@@ -257,11 +257,7 @@ public class EntityRobot extends EntityRobotBase implements
 
 			if (energyFX >= 100) {
 				energyFX = 0;
-				Minecraft.getMinecraft().effectRenderer.addEffect(new EntityRobotEnergyFX(
-						worldObj,
-						posX + steamDx * 0.25, posY + steamDy * 0.25, posZ + steamDz * 0.25,
-						steamDx * 0.05, steamDy * 0.05, steamDz * 0.05,
-						energySpendPerCycle * 0.075F < 1 ? 1 : energySpendPerCycle * 0.075F));
+				spawnEnergyFX();
 			}
 		}
 
@@ -316,6 +312,14 @@ public class EntityRobot extends EntityRobotBase implements
 		}
 
 		super.onEntityUpdate();
+	}
+	@SideOnly(Side.CLIENT)
+	private void spawnEnergyFX() {
+	    Minecraft.getMinecraft().effectRenderer.addEffect(new EntityRobotEnergyFX(
+                worldObj,
+                posX + steamDx * 0.25, posY + steamDy * 0.25, posZ + steamDz * 0.25,
+                steamDx * 0.05, steamDy * 0.05, steamDz * 0.05,
+                energySpendPerCycle * 0.075F < 1 ? 1 : energySpendPerCycle * 0.075F));        
 	}
 
 	public void setRegularBoundingBox() {

